### DISCUSSION TITLE
Can it be possible inject "windows_injection.7z" files to Windows ISO's WinPE section?

### DISCUSSION THREAD

**Discussion by crasadure** [ventoy/Ventoy#1159] [Ideas]
hi @ventoy 
I tried and succeed  to inject "windows_injection.7z" files into Windows ISO boot.wim file's Second image section(2: İnstall index)
you can watch my video from here : https://youtu.be/bwYkM4iaC6o

### First Question!!
Now, if I want to inject these files to WinPE section of boot.wim file, how can we add this files to WinPE section?can it be possible guy?
![image1](https://user-images.githubusercontent.com/54566818/137956836-2f8296cc-38ad-40f6-b5d6-42323a03acd3.jpg)

### Second Question!!
Can we inject "windows_injection.7z" files  into **boot.wim** file?(NOT inject boot.wim file in Windows ISO , only one file of windows orjinal **boot.wim** file)
I want to make that , I exctracted  the **boot.wim** file from the standart Windows ISO file and I want to INJECT "_windows_injection.7z_" files into the **boot.wim** file by using ventoy injection plugin.
Watch my video pls! Video Link : https://vimeo.com/636521413

* can we inject "windows_injection.7z" files in BOOT.WIM file, is it possible with Ventoy İnjection Plugin? 
* can you add this property to Injection Plugin,guy?

### Third Question!!
Can we start  standart windows ISO from WinPE section ?
Grub2 file manager program can load and start Windows ISO's like as WinPE  
![g2fm_start winpe from windows iso](https://user-images.githubusercontent.com/54566818/137961005-6ed8c523-a8f3-48c4-9509-b41c3387be9e.jpg)
**OR** 
Can load and start Windows ISO like as **_Windows installation ISO_**
![g2fm_start windows](https://user-images.githubusercontent.com/54566818/137961236-72e639ea-ad05-4988-afc9-5f8a63db1b93.jpg)

Grub2 file manager are giving to users chance choosing a choice (Load Winpe OR İnstall Windows from Windows ISO) from standart  Windows ISO file
Watch video pls : https://vimeo.com/636558779
Can it be posssible for Ventoy multiboot?

**Comment by ventoy** [comment]
Q1:
According to WIM file spec, a wim file can only have one bootable wim index.
How can you boot into index 1 ?

Q2:
Currently, injection for bare wim file is not supported. You need to wrapp it as an ISO file first.

Q3:
Grub2FileManager `Boot Windows PE From ISO` feature actually does similar thing as Ventoy injection plugin.
https://github.com/a1ive/grub2-filemanager/blob/4a16f24e0f138e9b96418fe9637e979fb3e0dc69/boot/grubfm/rules/iso/buildpe.sh#L16
It just use wimboot to inject a `winpeshl.ini` and `sfx64.exe (winxshell.exe)` to the wim file, so after boot, `sfx64.exe(winxshell.exe)` is launched instead of `setup.exe`. So it seems that you boot into a PE.
In this case, still index 2 is used not index 1 (you can see X:\setup.exe which only exist in index 2)
